### PR TITLE
made fishing.py a bit easier to look at

### PR DESCRIPTION
**Pull request by iHeroGH**

### CODE REVIEW

File: cogs/fishing.py [hunk 1]
```diff
@@ -227,21 +222,25 @@ async def fish(self, ctx: commands.Context):
...
-        elif special == "golden":
-            new_fish = utils.make_golden(new_fish)
+        
+        special_functions = {
+            "inverted": utils.make_inverted(new_fish),
+            "golden": utils.make_golden(new_fish)
+        }
+        
+        if special in special_functions.keys():
+            new_fish = special_functions[special]
```
Comment (4Kaylum) [inline cogs/fishing.py:232]: This won't work - it'll run `make_inverted` and `make_golden` every time. If you don't want that then you should wrap it in a function and call it as necessary, but I don't think this improves readability anyway.

File: cogs/fishing.py [hunk 2]
```diff
@@ -290,30 +289,36 @@ async def rename(self, ctx: commands.Context, old: str, new: str):
         """
         Renames your fish.
         """
-
-
+
```
Comment (4Kaylum) [inline cogs/fishing.py:292]: Smh set up your VSC to trim trailing spaces.

**Review by 4Kaylum — CHANGES_REQUESTED** [review]

File: cogs/fishing.py (hunk 1 above)
Comment (iHeroGH) [reply cogs/fishing.py:232]: Why will it run it every time? Wouldn't it just run it if the special word is in the dict and then run just one of the functions? It's pretty similar to how it was before but this time with a dictionary

File: cogs/fishing.py (hunk 2 above)
Comment (iHeroGH) [reply cogs/fishing.py:292]: I did this in GitHub 😳

File: cogs/fishing.py (hunk 1 above)
Comment (4Kaylum) [reply cogs/fishing.py:232]: The dictionary gives a value, not a "state". `inverted` will have the value returned by `utils.make_inverted`, for example, and to get the value the function needs to be run.

image

Comment (iHeroGH) [reply cogs/fishing.py:232]: new_fish should be the value returned by make_inverted or make_golden.

the make_... commands return a dict where the name of the fish is either "inverted_fishname" or golden.

Either this does that or I'm very confused lol

Comment (4Kaylum) [reply cogs/fishing.py:232]: Ah. Yes you're normally right. These functions *modify* the input dictionary though, not just return a new one.

**Pull request merged**